### ISSUE TITLE
Empty model after deleting orphaned element

### ISSUE THREAD

**Issue by Sycophantic-Witty** [gaphor/gaphor#2731]
### Describe the bug
Deleting an orphaned and nameless element appearing at the top of the model hierarchy and saving the model causes Gaphor to no longer display anything when reloading the model.

How this orphaned element ended here, I cannot say. I did some create/delete/undo/redo actions, and saved the model that looked ok. Not 100% sure but I think this orphaned element appeared only after reloading the model.

### To Reproduce
1. Load the attached model.
2. Right click on the ': First Child' element in the Model Browser and select 'Delete'.
3. Save the model.
4. Restart Gaphor and load the model again.
5. Gaphor displays an empty Model Browser.

Error Reports after reopening the model:
- Windows
  Gaphor version:         2.21.0
  Operating System:       Windows (10)
  Display:                GdkWin32Display
  Python version:         3.11.5
  GTK version:            4.12.0
  Adwaita version:        1.3.1
  GtkSourceView version:  5.8.0
  Cairo version:          1.17.8
  Pango version:          1.50.14
  PyGObject version:      3.44.1
  Pycairo version:        1.24.0
  pygit2/libgit2 version: 1.12.2  / 1.6.4

  No errors have been intercepted. We’re good.

- Debian GNU/Linux
  Gaphor version:         2.21.0
  Operating System:       Linux (5.15.90.1-microsoft-standard-WSL2)
  Display:                GdkWaylandDisplay
  Python version:         3.11.2
  GTK version:            4.8.3
  Adwaita version:        1.2.2
  GtkSourceView version:  -
  Cairo version:          1.16.0
  Pango version:          1.50.12
  PyGObject version:      3.44.1
  Pycairo version:        1.24.0
  pygit2/libgit2 version: 1.12.2  / 1.6.4

  No errors have been intercepted. We’re good.

### Expected behavior
After reloading the model, I expect to see the same structure as that displayed when saving the model.

### Screenshots
No screenshot, issue easily reproduceable with the attached model.

### OS
- [X] Linux (Debian 12.1 under WSL2)
- [ ] macOS
- [X] Windows (using the released Windows installer)

### Version
Version of Gaphor: 2.21.0

### Additional information
Interestingly, when starting Gaphor (from the command-line) installed with `pip` in a Debian GNU/Linux system, Gaphor displays a dialog showing the error message "Unable to open model ..." after deleting the element and reloading the model. The Windows version does not show any error dialog.

[uml-test-model.zip](https://github.com/gaphor/gaphor/files/12767831/uml-test-model.zip)

**Comment by amolenaar** [comment]
This issue looks similar to #2707.

Stack trace:

```
(Pdb) bt
  /home/arjan/Development/gaphor/gaphor/ui/filemanager.py(213)_load_async()
-> for percentage in storage.load_generator(
  /home/arjan/Development/gaphor/gaphor/storage/storage.py(307)load_generator()
-> for percentage in load_elements_generator(
  /home/arjan/Development/gaphor/gaphor/storage/storage.py(171)load_elements_generator()
-> elem.element.postload()
  /home/arjan/Development/gaphor/gaphor/UML/classes/association.py(133)postload()
-> super().postload()
  /home/arjan/Development/gaphor/gaphor/diagram/presentation.py(350)postload()
-> postload_connect(self, self.tail, self._load_tail_connection)
  /home/arjan/Development/gaphor/gaphor/diagram/presentation.py(72)postload_connect()
-> connect(item, handle, target)
  /home/arjan/Development/gaphor/gaphor/diagram/presentation.py(61)connect()
-> connector.connect(sink)
  /home/arjan/Development/gaphor/gaphor/transaction.py(160)_transactional()
-> return func(*args, **kwargs)
  /home/arjan/Development/gaphor/gaphor/diagram/_connector.py(70)connect()
-> adapter.connect(handle, sink.port)
  /home/arjan/Development/gaphor/gaphor/diagram/connectors.py(314)connect()
-> self.connect_subject(handle)
  /home/arjan/Development/gaphor/gaphor/UML/classes/classconnect.py(149)connect_subject()
-> relation = self.relationship_or_new(head_end.subject, tail_end.subject)
  /home/arjan/Development/gaphor/gaphor/UML/classes/classconnect.py(123)relationship_or_new()
-> relation.memberEnd[1].type = tail_subject
> /home/arjan/Development/gaphor/gaphor/core/modeling/collection.py(42)__getitem__()
-> return self.items.__getitem__(key)
```

When i diff the the model with the newly saved one, after deleting the `: First Child` element, I see:

```diff
$ diff uml-test-model-orig.gaphor uml-test-model.gaphor
428,430d427
< <head_subject>
< <ref refid="108d275a-5dd5-11ee-ad92-489ebdd5189b"/>
< </head_subject>
459d455
< <ref refid="108d275a-5dd5-11ee-ad92-489ebdd5189b"/>
472,479d467
< <Property id="108d275a-5dd5-11ee-ad92-489ebdd5189b">
< <association>
< <ref refid="108cd94f-5dd5-11ee-a7df-489ebdd5189b"/>
< </association>
< <type>
< <ref refid="607b863c-53c3-11ee-aa4f-00155d86fc47"/>
< </type>
< </Property>
```

An Association is supposed to have two properties attached, one for each end. However, by removing that property, it also broke the association.

Question is: why isn't the property showing up on top level, and not as a child of the association.

Thanks @Sycophantic-Witty, this is very helpful.

**Comment by amolenaar** [comment]
Another issue that a property ends up at top level if the navigability of an association end is set to "Not Navigable".

Not sure if implementing #633 (UML 2.5 Association syntax) will fix this.

**Comment by Sycophantic-Witty** [comment]
> Another issue that a property ends up at top level if the navigability of an association end is set to "Not Navigable".

Great, thanks! Reverting its head to **Unknown navigation** moves the property back, below the relationship.

**Comment by amolenaar** [comment]
The problem is with how navigability is managed. 

Always `Association.navigableOwnedEnd` is set. However, this property is a `member`, not an `owner`.

Currently, depending on the navigability setting the ownership is different:

| Navigability  | Property set |
| ------------- | ------------- |
| Navigable |   owned by `Type.ownedAttribute` |
| Not Navigable | `Association.navigableOwnedEnd` set (no ownership) |
| Unknown Navigability | set `Association.ownedEnd` (with ownership) |

One idea is to check `NamedElement.member` as a fallback in the Model Browser, if `Element.owner` is not set.
Also we should check if the current implementation follows the UML 2.5 semantics.

**Comment by danyeaw** [comment]
@all-contributors please add @Sycophantic-Witty for bug

**Comment by danyeaw** [comment]
Thanks @Sycophantic-Witty, @amolenaar's change should significantly reduce orphaned elements.
